### PR TITLE
Temporarily freeze react version, upgrade other dependencies

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Temporarily freeze react version at v18.2.0 until we have a chance to do that upgrade (tracked here: https://github.com/grommet/grommet/issues/7127).

This PR upgrades some other dependencies

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap [hunk 1]
```diff
@@ -464,8 +464,8 @@ exports[`Anchor matches icon size to size prop when theme.icon.matchSize is true
   -webkit-flex: 0 0 auto;
   -ms-flex: 0 0 auto;
   flex: 0 0 auto;
-  width: NaNpx;
-  height: NaNpx;
+  width: NaNxsmall;
```
Comment (britt6612) [inline src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:467]: do we want these? I remember they are from `grommet-icons` wasn't sure if we looked into why they changed

Comment (jcfilben) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:467]: I'm guessing it is coming from this PR, but not positive: https://github.com/grommet/grommet-icons/pull/325/files
Either way I think visually this test wouldn't change. The width is just being replaced with another invalid option.
At some point we should look at the `useSizedIcon` function in Grommet, I think this is where the issue may be coming from. I think the icons are supposed to scale to the size of the Anchor. This isn't working currently and this PR doesn't change that it just changes the incorrect value to another incorrect value.
Screen Shot 2024-05-14 at 9 00 58 AM

Comment (taysea) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:467]: Do we have a ticket somewhere to capture?

Comment (jcfilben) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:467]: Issue created: https://github.com/grommet/grommet/issues/7231

**Review by taysea — APPROVED** [review]

**Pull request merged**